src/components/WalletProvider/index.svelte

```diff
@@ -173,12 +173,15 @@
     };
 
     let requestConnection = async () => {
-        await requestAddEthereumChain(network).catch((_) => {
-            if (parseInt($state.chainId) !== parseInt(chainId)) {
-                setState({ connectionError: errorCodes.WRONG_CHAIN });
-                return Promise.reject($state.connectionError);
+        await requestAddEthereumChain(network).catch((err) => {
+            if (err.code !== 4902) {
```

Comment:
else?

Reply:
@zolotokrylin There is no else condition, it is handled by the function `requestAddEthereumChain`